spectator/id.py

```diff
@@ -31,3 +31,10 @@ def __str__(self):
         items = sorted(self._tags.items())
         tagStr = ",".join(["{}={}".format(t[0], t[1]) for t in items])
         return "{}:{}".format(self.name, tagStr)
+
+    @staticmethod
+    def sanitize(tags):
+        t = {}
+        for k, v in tags.items():
+            t[str(k)] = str(v)
```

Comment:
The concern I have is that many objects have a string conversion that is not appropriate for tags and it makes it really easy to high cardinality garbage. For example, if they added a file handle it would give something like:

```
<open file '/tmp/foo', mode 'r' at 0x10faada50>
```

The address portion of that could blow up rather quickly. Perhaps a better option is to drop the metric and log warning with the full set of tags and saying why it was dropped.

Reply:
I like this idea - it's transparent for the developer and it helps avoid an issue that could easily arise in code, given the dynamic nature of the language.